nautobot/docs/development/apps/migration/ui-component-framework.md

```diff
@@ -0,0 +1,244 @@
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['related_devices'] = self.object.related_devices.all()
+        context['custom_fields'] = self.object.get_custom_fields()
+        return context
+
+# template.html
+{% extends 'base.html' %}
+{% block content %}
+<div class="row">
+    <div class="col-md-6">
+        <div class="panel">
+            <!-- Manual HTML for device fields -->
+        </div>
+    </div>
+    <div class="col-md-6">
+        <!-- More manual HTML -->
+    </div>
+</div>
+{% endblock %}
+```
+
+After (UI Framework):
+```python
+from nautobot.core.views import ObjectView
+from nautobot.core.views.content import ObjectDetailContent
+from nautobot.core.views.panels import (
+    ObjectFieldsPanel,
+    ObjectsTablePanel,
+    TextPanel,
+)
+
+class DeviceDetailView(ObjectView):
+    queryset = Device.objects.all()
+    
+    object_detail_content = ObjectDetailContent(
+        panels=[
+            ObjectFieldsPanel(
+                fields="__all__",
+                section=SectionChoices.LEFT_HALF,
+                weight=100,
+            ),
+            ObjectsTablePanel(
+                table_class=RelatedDeviceTable,
+                table_attribute="related_devices",
+                section=SectionChoices.RIGHT_HALF,
+                weight=100,
+            ),
+        ]
+    )
+```
+
+## Migration Steps
+
+### 1. Identify View Components
+First, analyze your existing view to identify its components:
+
+- Object fields display
+- Related object tables
+- Custom field groups
+- Statistics/counts
+- Text content (descriptions, notes)
+- Custom template sections
+
+### 2. Choose Appropriate Panels
+
+Map your existing components to UI Framework panels:
+
+| Current Feature | UI Framework Panel |
+|----------------|-------------------|
+| Basic field display | `ObjectFieldsPanel` |
+| Related object tables | `ObjectsTablePanel` |
+| Key-value data | `KeyValueTablePanel` |
+| Grouped fields | `GroupedKeyValueTablePanel` |
+| Statistics | `StatsPanel` |
+| Markdown/text content | `TextPanel` or `ObjectTextPanel` |
+
+### 3. Convert Views
+
+1. Update your view class:
+```python
+# Before
+class MyDetailView(generic.ObjectView):
+    template_name = 'myapp/detail.html'
+
+# After
+class MyDetailView(ObjectView):
+    object_detail_content = ObjectDetailContent(
+        panels=[
+            # Panel definitions
+        ]
+    )
+```
+
+2. Move context data into panels:
+```python
+# Before
+def get_context_data(self, **kwargs):
+    context = super().get_context_data(**kwargs)
+    context['stats'] = self.get_stats()
+    return context
+
+# After
+StatsPanel(
+    filter_name="device",
+    related_models=[
+        Device,
+        (Circuit, "circuit_terminations__device__in")
+    ],
+    section=SectionChoices.RIGHT_HALF,
+    weight=100,
+)
+```
+
+### 4. Layout Organization
+
+Organize your panels using sections and weights:
```

Comment:
I couldn't find any explanation how sections and weights are work together. Maybe add just one statement that full width panels will be always under half's-versions and on top will be the panel with smallest weight?